src/mappings/utils/entity.ts

```diff
@@ -72,3 +72,37 @@ export function create<T extends EntityWithId>(
   Object.assign(entity, init);
   return entity;
 }
+
+export async function calculateCollectionOwnerCountAndDistribution(
+  store: Store,
+  collectionId: string,
+  newOwner?: string,
+  originalOwner?: string,
+): Promise<{ ownerCount: number; distribution: number }> {
+  const query = `
+  SELECT COUNT(DISTINCT current_owner) AS distribution,
+       COUNT(current_owner) AS owner_count
+  ${
+  newOwner
+    && `
+  ,(SELECT max(CASE
+                  WHEN current_owner = '${newOwner}' THEN 0
+                  ELSE 1
+              END)
+   FROM nft_entity) AS adjustment
+  `
+} 
+  FROM nft_entity
+  WHERE collection_id = '${collectionId}'
+  ${newOwner && `AND current_owner != '${originalOwner}'`}
+  `;
+  const queryResult: { owner_count: number; distribution: number; adjustment?: number }[] = await store.query(query);
+  const result = queryResult[0];
```

Comment:
This is something that I forgot to refactor
```suggestion
  const [result]: { owner_count: number; distribution: number; adjustment?: number }[] = await store.query(query);
```